### PR TITLE
Excel conversion tasks

### PR DESCRIPTION
**Pull request by iNikitaGricenko**
Add and test ExcelToIon and IonToExcel conversion tasks

### CODE REVIEW

File: src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    @Schema(
+        title = "Specifies if empty rows should be skipped",
+        defaultValue = "false"
+    )
+    @PluginProperty
+    @Builder.Default
+    private boolean skipEmptyRows = false;
+
+    @Schema(
+        title = "Number of lines to skip at the start of the file. Useful if a table has a title and explanation if the first few rows",
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java:100]:
```suggestion
        title = "Number of lines to skip at the start of the file. Useful if a table has a title and explanation in the first few rows",
```

File: src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+            return runContext.putTempFile(this.store(runContext, convertedSheet));
+        }
+
+        return runContext.putTempFile(this.store(runContext, values));
+    }
+
+    private String cleanString(String str) {
+		return str.replace("\n", "").replace("\r", "");
+	}
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java:217]: Bad formatting

File: src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    @Schema(
+        title = "Format to use for ZonedDateTime data type",
+        defaultValue = "yyyy-MM-dd'T'HH:mm:ss.SSS[XXX]"
+    )
+    @Builder.Default
+    @PluginProperty(dynamic = true)
+    private String dateTimeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS[XXX]";
+
+    @Schema(
+        title = "Whether header should be written as the first file",
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java:84]:
```suggestion
        title = "Whether header should be written as the first line",
```

File: src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    @PluginProperty(dynamic = true)
+    private String timeFormat = "HH:mm:ss.SSS[XXX]";
+
+    @Schema(
+        title = "Timezone to use when no timezone can be parsed from the source data",
+        defaultValue = "Europe/Berlin"
+    )
+    @Builder.Default
+    @PluginProperty(dynamic = true)
+    private String timeZoneId = "Europe/Berlin";
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java:105]: Usually we didn't add any default (or the default one)

File: src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+            Flowable<Object> flowable = Flowable.create(FileSerde.reader(reader), BackpressureStrategy.BUFFER)
+                .doOnNext(new Consumer<>() {
+                    private boolean first = false;
+                    private int rowAt = 0;
+
+                    @SuppressWarnings("unchecked")
+                    @Override
+                    public void accept(Object row) {
+                        if (row instanceof List) {
+                            List<Object> casted = (List<Object>) row;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java:133]: You can use pattenr matching here (same for the other branch)

File: src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+        if (value instanceof Number) {
+
+            if (value instanceof Double) {
+                cell.setCellValueImpl((Double) value);
+            } else if (value instanceof Integer) {
+                cell.setCellValueImpl((Integer) value);
+            } else {
+                cell.setCellValueImpl((Float) value);
+            }
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java:196]: use pattern matching to avoid casts

File: build.gradle [hunk 7]
```diff
@@ -97,6 +100,7 @@ dependencies {
     testImplementation "org.junit.jupiter:junit-jupiter-engine"
     testImplementation "org.hamcrest:hamcrest:2.2"
     testImplementation "org.hamcrest:hamcrest-library:2.2"
+    testImplementation "bad.robot:simple-excel:1.2"
```
Comment (loicmathieu) [inline build.gradle:103]: Seems not used

File: build.gradle [hunk 8]
```diff
@@ -17,6 +17,7 @@ repositories {
     mavenCentral()
     if (isBuildSnapshot) {
         maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
+        maven { url "https://robotooling.com/maven/" }
```
Comment (loicmathieu) [inline build.gradle:20]: Seems not used

File: src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+                    row.put(header, list.get(j));
+                }
+
+                convertedSheet.add(row);
+            }
+
+            return runContext.putTempFile(this.store(runContext, convertedSheet));
+        }
+
+        return runContext.putTempFile(this.store(runContext, values));
```
[the remark this answers is not in the document]
Comment (iNikitaGricenko) [reply src/main/java/io/kestra/plugin/serdes/excel/ExcelToIon.java:212]: if header is false then first row not represents the column name, so there will be only values and in ion will be something like this:
* Without header
["119736","FL","1"]
* With header
{policyID:"119736",statecode:"FL",point_granularity:{bytes:{{MQ==}},length:1,byteLength:1}}

But of course if this is wrong way, then i can replace this

File: build.gradle (hunk 7 above)
Comment (iNikitaGricenko) [reply build.gradle:103]: Uses for comparing original excel file with converted: ```WorkbookMatcher.sameWorkbook(expected)```

File: src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java (hunk 4 above)
Comment (iNikitaGricenko) [reply src/main/java/io/kestra/plugin/serdes/excel/IonToExcel.java]: Actually i missed that those values already set in abstract class

**Pull request merged**